### PR TITLE
[FABCJ-214] merged to release-2.2

### PR DESCRIPTION
**Pull request by kemi04**
Signed-off-by: Kestutis Gudynas <kestutis.gudynas@ibm.com>

Java Chaincode GRPC server commit back ported to _release-2.2_ branch

### CODE REVIEW

File: examples/fabric-contract-example-gradle/fabric-contract-example-external/build.gradle [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        url "https://www.jitpack.io"
+    }
+    maven {
+        url "https://hyperledger.jfrog.io/hyperledger/fabric-maven"
+    }
+
+}
+
+dependencies {
+    compile group: 'org.hyperledger.fabric-chaincode-java', name: 'fabric-chaincode-shim', version: '2.3.+'
```
Comment (mbwhite) [inline examples/fabric-contract-example-gradle/fabric-contract-example-external/build.gradle:23]: We should change this to the 2.2 version.

File: examples/fabric-contract-example-gradle/fabric-contract-example-external/src/main/java/org/example/Application.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+import java.io.IOException;
+import org.hyperledger.fabric.contract.ContractRouter;
+import org.hyperledger.fabric.shim.ChaincodeServer;
+import org.hyperledger.fabric.shim.ChaincodeServerProperties;
+import org.hyperledger.fabric.shim.NettyChaincodeServer;
+
+public class Application {
+
+    private static final String CHAINCODE_SERVER_PORT = "CHAINCODE_SERVER_PORT";
```
Comment (mbwhite) [inline examples/fabric-contract-example-gradle/fabric-contract-example-external/src/main/java/org/example/Application.java:15]: I've checked the example at https://github.ibm.com/WHITEMAT/contract-as-a-server/blob/main/java-contract/src/main/java/org/example/ContractBootstrap.java.

It's slightly different - in that in takes the full IP and port number to be consistent with node... Might be worth changing that here as well...

**Review by mbwhite — COMMENTED** [review]
Should be good:-)

File: examples/fabric-contract-example-gradle/fabric-contract-example-external/build.gradle (hunk 1 above)
Comment (kemi04) [reply examples/fabric-contract-example-gradle/fabric-contract-example-external/build.gradle]: Done

File: examples/fabric-contract-example-gradle/fabric-contract-example-external/src/main/java/org/example/Application.java (hunk 2 above)
Comment (kemi04) [reply examples/fabric-contract-example-gradle/fabric-contract-example-external/src/main/java/org/example/Application.java]: Done

**Review by mbwhite — APPROVED** [review]

**Pull request merged**